### PR TITLE
Implemented dataset pagination

### PR DESCRIPTION
**Pull request by ds161833**
image
Title says it all 😄 also created a List component that can be reused everywhere we use list->row pattern

### CODE REVIEW

File: Client/src/Components/Utils/List.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+interface IProps<T, ICustomRowProps> {
+  RowComponent: React.ComponentType<IRowProps<T>>
+  models: T[]
+  rowProps: ICustomRowProps
+  withPagination: boolean
+}
+
+export function List<T, ICustomRowProps>(props: IProps<T, ICustomRowProps>) {
```
Comment (laithAwad) [inline Client/src/Components/Utils/List.tsx:15]: I like this a lot! we use lists very often and this simplifies things

**Review by laithAwad — APPROVED** [review]
Great work! very clean

File: Client/src/Components/Graph/GraphControls/DatasetControl.tsx [hunk 2]
```diff
@@ -72,10 +73,11 @@ export const DatasetControl = (props: IProps) => {
...
       </Grid>
-      <DatasetList
-        datasets={toDatasetRows(completeDatasets, datasetStates)}
-        onRemoveDatasetClick={handleDatasetRemoved}
-        onHideDatasetSwitch={onHideDatasetSwitch}
+      <List
+        RowComponent={DatasetRow}
+        models={toDatasetRows(completeDatasets, datasetStates)}
+        rowProps={{ onRemoveDatasetClick: handleDatasetRemoved, onHideDatasetSwitch: onHideDatasetSwitch } as IDatasetRowProps}
+        withPagination
```
Comment (ratulkai93) [inline Client/src/Components/Graph/GraphControls/DatasetControl.tsx:80]: looks very nice and clean. amazing work!!

File: Client/src/Components/Utils/List.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        labelRowsPerPage='Datasets per page'
+        count={models.length}
+        page={page}
+        onChangePage={(_, newPage) => setPage(newPage)}
+        rowsPerPage={rowsPerPage}
+        onChangeRowsPerPage={(event) => setRowsPerPage(Number(event.target.value))}
+      />}
+    </>
+  )
+}
```
Comment (ratulkai93) [inline Client/src/Components/Utils/List.tsx:41]: good stuff

**Review by ratulkai93 — APPROVED** [review]
Everything looks very nice and clean. Thank you for your hard work!

**Pull request merged**